### PR TITLE
ParaView: Make build edition strings uppercase

### PR DESCRIPTION
**Pull request by EthanS94**
@danlipsa It seems the cmake argument for ParaView build_editions expects the strings to be all uppercase, at least for `canonical`

Without capitalization, cmake was ignoring arguments that have worked in the past:

```console
  Manually-specified variables were not used by the project:

    MPI_Fortran_COMPILER
    OpenGL_GL_PREFERENCE
    PARAVIEW_ENABLE_EXAMPLES
    PYTHON_EXECUTABLE
    VTK_MODULE_USE_EXTERNAL_ParaView_cgns
    VTK_MODULE_USE_EXTERNAL_VTK_gl2ps
    VTK_MODULE_USE_EXTERNAL_VTK_glew
    VTK_MODULE_USE_EXTERNAL_VTK_libharu
    VTK_OPENGL_HAS_OSMESA
```

With capitalization, at least for `canonical`, many of these are set as expected:

```console
  Manually-specified variables were not used by the project:

    MPI_Fortran_COMPILER
    PARAVIEW_ENABLE_EXAMPLES
    PYTHON_EXECUTABLE


-- Build files have been written to: /tmp/stam/spack-stage/spack-stage-paraview-5.9.1-ydbfmizfhppij5vjhrfnjsuo3eni3olg/spack-build-ydbfmiz
```

This PR adds `.upper()` to the string passed in the cmake argument.

### CODE REVIEW

**Comment by EthanS94** [comment]
@DarylGrunau I think this will fix your issue.

**Comment by DarylGrunau** [comment]
Testing this out now - thanks!

**Comment by DarylGrunau** [comment]
This fixes my issue - thanks @EthanS94 !

**Comment by DarylGrunau** [comment]
Just pinging @chuckatkins, @danlipsa and @vicentebolea again about this bug.  IMHO the fix is so simple and clearly meets the need for proper CMake defines for the various PV Build Editions; perhaps a maintainer need not get involved at all?  @adamjstewart, would you support this?  Not having this fix in place is blocking me from releasing a new Spack Environment.

**Review by vicentebolea — APPROVED** [review]
LGTM +1

**Review by adamjstewart — APPROVED** [review]

**Comment by DarylGrunau** [comment]
@adamjstewart would you be willing to just merge this?  We have one approval from a maintainer ...  MANY THANKS!

**Pull request merged**
